### PR TITLE
Hashdump - minor fixes

### PR DESCRIPTION
**Pull request by otaviobp**

### CODE REVIEW

File: src/hashdump.c [hunk 1]
```diff
@@ -102,36 +97,33 @@ int hashdump_main(int argc, char **argv)
...
 	} else {
-		printf("%s\n", file->hash);
-		if (file->is_dir) {
-			if (is_directory_mounted(fullname)) {
-				fprintf(stderr, "!! dumped hash might not match a manifest hash because a mount is active\n");
-			}
+		printf("%s\n", file.hash);
+		if (file.is_dir && is_directory_mounted(fullname)) {
+			fprintf(stderr, "!! dumped hash might not match a manifest"
+					"hash because a mount is active\n");
```
Comment (jwakre) [inline src/hashdump.c]: I think this will print: "manifesthash" instead of "manifest hash"

Comment (otaviobp) [reply src/hashdump.c]: thanks. fixed.

File: src/hashdump.c [hunk 2]
```diff
@@ -83,35 +83,30 @@ int hashdump_main(int argc, char **argv)
...
-	ret = set_path_prefix(NULL);
-	if (!ret) {
-		free_string(&file->filename);
+	if (!set_path_prefix(NULL)) {
 		free(file);
-		return EXIT_FAILURE;
+		return EINIT_GLOBALS;
 	}
 
+	file.filename = strdup_or_die(argv[optind]);
```
Comment (jwakre) [inline src/hashdump.c]: I think this line should be moved to the commit that changes file to static allocation.

Comment (otaviobp) [reply src/hashdump.c]: fixed.

**Pull request merged**